src/inc-lib-init.php

```diff
@@ -1084,336 +1084,488 @@ function lovd_getInstallURL ($bFull = true)
+        'earliest_intronic_start' => (!isset($aMatches[6])?   0 : (int)str_replace('?', '1', $aMatches[6])),
+        'latest_start'            => (!isset($aMatches[9])?   0 : $aMatches[9]),
+        'latest_intronic_start'   => (!isset($aMatches[10])?  0 : (int)str_replace('?', '1', $aMatches[10])),
+        'earliest_end'            => (!isset($aMatches[14])?  0 : $aMatches[14]),
+        'earliest_intronic_end'   => (!isset($aMatches[15])?  0 : (int)str_replace('?', '1', $aMatches[15])),
+        'latest_end'              => (!isset($aMatches[17])?  0 : $aMatches[17]),
+        'latest_intronic_end'     => (!isset($aMatches[18])?  0 : (int)str_replace('?', '1', $aMatches[18])),
+        'type'                    => (!isset($aMatches[20])? '' : $aMatches[20]),
+        'suffix'                  => (!isset($aMatches[23])? '' : $aMatches[23]),
+    ));
+
+    if (!isset($aVariant['complete']) || $aVariant['complete'] != $sVariant) {
+        // If the complete match from getHGVSpatternFromVariant is not set or does not equal the given variant,
+        //  the variant is not HGVS, and we cannot extract any information from it.
+        if ($bCheckHGVS) {
+            return false;
+        }
+        foreach (array('qter', 'pter', 'cen', '::', 'sup', 'bsr', 'rer') as $sUnsupported) {
+            if (strpos($sVariant, $sUnsupported)) {
+                $aResponse['errors']['ENOTSUPPORTED'] =
+                    'Currently, "' . $sUnsupported . '" is not yet supported by this HGVS check.';
+                return $aResponse;
+            }
+        }
+        return false;
+    }
+
+
+    // Storing the variant type.
+    if (!$aVariant['type']) {
+        // If no type is given, we can be sure that the variant is either a full wildtype or a full
+        //  unknown variant; so either g.= or g.? . In this case, we do not need to go over all tests.
+        if (in_array($aVariant['prefix'], array('c', 'n'))) {
+            $aResponse['position_end_intron'] = 0;
+            $aResponse['position_start_intron'] = 0;
+        }
+        $aResponse['type'] = (substr($aVariant['complete'], 2, 3) == '='? '=' : NULL);
+        return $bCheckHGVS? true : $aResponse;
+        
+        
+    } elseif (strpos($aVariant['type'], '>')) {
+        $aResponse['type'] = 'subst';
+        
+    } elseif (substr($aVariant['type'], -1) == ']') {
+        $aResponse['type'] = 'repeat';
+        $aResponse['warnings']['WNOTSUPPORTED'] =
+            'Repeat variants are currently not supported for mapping and validation.';
+        
+    } elseif ($aVariant['type'][0] == '|') {
+        $aResponse['type'] = 'met';
+
+    } elseif ($aVariant['type'] == '?') {
+        $aResponse['type'] = NULL;
+
+    } elseif ($aVariant['type'] == '=') {
+        $aResponse['type'] = '=';
+
+    } else {
+        $aResponse['type'] = $aVariant['type'];
+    }
```

Comment:
Na Johan's eerdere opmerking vandaag, realiseer ik me dat we geen mosaic en chimeric varianten ondersteunen. Kunnen we die makkelijk toevoegen, denk je?